### DISCUSSION TITLE
path name as an integer ???

### DISCUSSION THREAD

**Discussion by rmyrz** [bluenviron/mediamtx#865] [Questions] [answered]
Hi,
I have RTSP simple server v0.17.17
Does there need to be a conversion from the "path" to read it as a literal string and not an integer?  Whenever I have a path that is an integer, it gives me the below error

Example:
  **60067:
    source: rtsp://x.y.w.z:8554/swVideo
    sourceOnDemand: yes**

$ ./rtsp-simple-server 
panic: interface conversion: interface {} is int, not string

goroutine 1 [running]:
github.com/aler9/rtsp-simple-server/internal/conf.loadFromFile.func1({0xaa0ba0, 0xc0002dc4e0})
        /s/internal/conf/conf.go:75 +0x213
github.com/aler9/rtsp-simple-server/internal/conf.loadFromFile.func1({0xaa0ba0, 0xc0002dc180})
        /s/internal/conf/conf.go:75 +0x18b
github.com/aler9/rtsp-simple-server/internal/conf.loadFromFile({0xb70b02, 0x16}, 0x40d3e7)
        /s/internal/conf/conf.go:89 +0x1a9
github.com/aler9/rtsp-simple-server/internal/conf.Load({0xb70b02, 0x16})
        /s/internal/conf/conf.go:210 +0x45
github.com/aler9/rtsp-simple-server/internal/core.New({0xc0000301e0, 0x0, 0x0})
        /s/internal/core/core.go:83 +0x5ce
main.main()
        /s/main.go:10 +0x54

**Comment by aler9** [comment, marked as answer]
Thanks for reporting this issue, the panic has been fixed, but anyway you shouldn't use integer as keys, you should use quoted integers:

```yaml
paths:
  '60067':
    source: rtsp://x.y.w.z:8554/swVideo
    sourceOnDemand: yes
```

**Reply by rmyrz** [reply]
Thank you so much for your hard work on this project!!!

[a comment by an automated account is not shown]
